File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
#include <stack>
class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        // Base Case
        vector<int> inOrder;
        if (root == NULL)
        {
            return inOrder;
        }
        if (root->left == NULL && root->right == NULL) {
            inOrder.push_back(root->val);
            return inOrder;
        }
         
        // Get left skewed Tree
        stack<TreeNode*> myLeftTree;
        TreeNode* leftNode = root;
        while (leftNode != NULL) {
            myLeftTree.push(leftNode);
            leftNode = leftNode->left;
        }
        
        //Start popping the elements

        while(!myLeftTree.empty()) {
            TreeNode* node = myLeftTree.top();
            myLeftTree.pop();
            inOrder.push_back(node->val);
            if (node->right != NULL) {
                
                TreeNode* rightNode = node->right;
                while (rightNode != NULL) {
                    myLeftTree.push(rightNode);
                    rightNode = rightNode->left;
                }
            }
        }
        return inOrder;
    }
};

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        MyLeftTree = list()
        currNode = root
        MyLeftTree.append(currNode)
        while(currNode.left != None):
            MyLeftTree.append(currNode.left)
            currNode = currNode.left
        
        while(len(MyLeftTree) != 0):
            currNode = MyLeftTree.pop()
            if (k == 1):
                return currNode.val;
            k-=1
            if (currNode.right!= None):
                MyLeftTree.append(currNode.right)
                currNode = currNode.right
                while(currNode.left != None):
                    MyLeftTree.append(currNode.left)
                    currNode = currNode.left
            